src/NetPanzer/Classes/Network/NetMessageDecoder.cpp

```diff
@@ -46,17 +46,18 @@ Uint16 NetMessageDecoder::decodeMessage(NetMessage** message) {
     return 0;  // no more messages
   }
 
-  Uint16* mlen = (Uint16*)(decode_message.data + offset);
-  unsigned int msg_len = ltoh16(*mlen);
+  Uint16 mlen_value;
+  memcpy(&mlen_value, decode_message.data + offset, sizeof(Uint16));
+  unsigned int msg_len = ltoh16(mlen_value);
 
   if (msg_len > size - sizeof(NetMessage) - offset) {
     LOGGER.warning("Malformed Multimessage!!");
     return false;
   }
 
-  *message = (NetMessage*)(mlen + 1);
+  *message = (NetMessage*)(decode_message.data + offset + sizeof(Uint16));
```

Comment:
Hey, that's awesome @winrid . I was looking at this code last night, I consulted with chatgpt, but I couldn't figure out the proper way. Nice patch!

Reply:
Thanks! I was confused too as I changed lines 49/50 above but it would still crash. So I sent the whole function to ChatGPT and it's like what about replacing these two magic numbers? then that worked :D

I think actually the bug is the + 1 in the old version instead of 2.